lib/creator.ex

```diff
@@ -0,0 +1,74 @@
+defmodule WiseHomex.Creator do
+  @moduledoc """
+  Auto creation of main wise_homex module functions from configuration
+
+  These function simply delegate to the api_client() function, which in turn sends a request to either
+  the API client implementation or the mock server
+  """
+
+  @doc """
+  When used by a module, we inject autogenerated mock functions from the api client definition
+  """
+  defmacro __using__([]) do
+    WiseHomex.ApiDefinition.api_definition()
+    |> WiseHomex.ApiDefinition.to_api_resources()
+    |> Enum.map(&WiseHomex.Creator.create_endpoint/1)
+  end
+
+  @doc """
+  Create a mock index function
+  """
+  def create_endpoint({:index, api_resource}) do
+    function_name = "get_#{api_resource.name_plural}" |> String.to_atom()
+
+    quote do
+      def unquote(function_name)(config, query \\ %{}), do: api_client().unquote(function_name)(config, query)
+    end
+  end
+
+  @doc """
+  Create a mock show function
+  """
+  def create_endpoint({:show, api_resource}) do
+    function_name = "get_#{api_resource.name_singular}" |> String.to_atom()
+
+    quote do
+      def unquote(function_name)(config, id, query \\ %{}), do: api_client().unquote(function_name)(config, id, query)
+    end
+  end
+
+  @doc """
+  Create a mock create function
+  """
+  def create_endpoint({:create, api_resource}) do
+    function_name = "create_#{api_resource.name_singular}" |> String.to_atom()
+
+    quote do
+      def unquote(function_name)(config, attrs, rels),
```

Comment:
Perhaps make `rels` optional to be consistent with `:update`?

Reply:
Good catch. I fixed it in all 3 creators, https://github.com/wise-home/wise_homex/pull/107/commits/d93034ee76ce8a22dfacdaff71d95c45fe70d6e3